File: app/src/main/java/com/example/bangabandhuplay/ui/movies/MoviesSliderAdapter.java
package com.example.bangabandhuplay.ui.movies;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.bangabandhuplay.data.model.sliders.Original;
import com.example.bangabandhuplay.databinding.MoviesSliderLayoutBinding;
import com.squareup.picasso.Picasso;


import java.util.List;

public class MoviesSliderAdapter extends RecyclerView.Adapter<MoviesSliderAdapter.mViewHolder> {
    private final List<Original> images;

    public MoviesSliderAdapter(List<Original> images) {
        this.images = images;
    }

    @NonNull
    @Override
    public MoviesSliderAdapter.mViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new MoviesSliderAdapter.mViewHolder(MoviesSliderLayoutBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull MoviesSliderAdapter.mViewHolder holder, int position) {
        Original current = images.get(position);
        String title = current.getTitle();
        String description = current.getDescription().toString();
        Picasso.get().load(current.getImage()).into(holder.binding.moviesSliderImage);
        if (title != null) holder.binding.sliderTitle.setText(title);
        if (description != null) holder.binding.sliderDesc.setText(description);
    }

    @Override
    public int getItemCount() {
        return images != null ? images.size(): 0;
    }

    public class mViewHolder extends RecyclerView.ViewHolder {
        MoviesSliderLayoutBinding binding;
        public mViewHolder(@NonNull MoviesSliderLayoutBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }
    }
}

File: app/src/main/java/live/durbar/bangabandhuapp/ui/movies/MoviesSliderAdapter.kt
package live.durbar.bangabandhuapp.ui.movies

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import live.durbar.bangabandhuapp.data.model.sliders.Original
import live.durbar.bangabandhuapp.databinding.MoviesSliderLayoutBinding
import live.durbar.bangabandhuapp.ui.player.PlayerActivity
import live.durbar.bangabandhuapp.utils.Constants
import com.squareup.picasso.Picasso

class MoviesSliderAdapter(private val images: List<Original>, private val context: Context) :
    RecyclerView.Adapter<MoviesSliderAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            MoviesSliderLayoutBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val current = images[position]
        val image = current.image
        val uuid = current.contentUrl

        image?.let { Picasso.get().load(it).into(holder.binding.moviesSliderImage) }

        holder.binding.root.setOnClickListener {
            if (uuid != null && uuid.isNotEmpty()) {
                context.startActivity(
                    Intent(context, PlayerActivity::class.java)
                        .putExtra(Constants.CONTENT_UUID, uuid)
                        .putExtra(Constants.CONTENT_SECTION_TITLE, "Slider")
                )
            }
            Constants.IS_MORE_CONTENT = false
            Constants.IS_MORE_HOME = false
        }
    }

    override fun getItemCount(): Int {
        return images.size
    }

    class ViewHolder(val binding: MoviesSliderLayoutBinding) : RecyclerView.ViewHolder(binding.root)
}
